### DISCUSSION TITLE
SVG paths to CrossSection

### DISCUSSION THREAD

**Discussion by Loosetooth** [elalish/manifold#506] [General]
Has anyone tried to convert SVG paths to `CrossSection` in order to use them in manifold? If so, how did you do it?

I was just looking a bit at the manifold JS bindings, and it looks like the CrossSection class can only be constructed from a list of points. Which is fine, but then it begs the question how SVG paths could be converted into a list of points correctly.

I had a look at the existing [jscad-text](https://github.com/jscad-community/jscad-text/blob/045ff2560a34556da52af2c6833264cdde124d82/src/textToPaths.js#L46C1-L121C2) package that has a function to transform SVG path commands into a JSCAD path.

Using (or drawing inspiration from) that plugin could be one way to do it.

The end goal for me here would be to be able to use text paths coming from Opentype.js and extrude them in manifold.

**Comment by pca006132** [comment]
Yeah I think we need to use something like that plugin. You can write some code to generate the Bezier interpolation, and then run `Path.Simplify` to simplify the paths. For the js binding, you can look at `glMatrix.vec3.bezier` which provides cubic bezier (but nothing else). Note that you will probably run into #502, which will need some time to fix.

**Reply by elalish** [reply]
Agreed - the fundamental difference is that Manifold deals only with polygons (straight sides), where SVG does curves. So you'll always need to decide at what resolution you want to convert, which is arbitrary and dependent on your needs.

**Reply by Loosetooth** [reply]
I managed to do it using Opentype.js and parts of the code of the jscad-text package. It seems like the code snippet from jscad-text that I linked to does both cubic and quadratic bezier interpolation.
